internal/provider/resource_group.go

```diff
@@ -0,0 +1,322 @@
+		byID = ptr.String(state.ID.ValueString())
+	}
+	if state.FullPath.ValueString() != "" {
+		byPath = ptr.String(state.FullPath.ValueString())
+	}
+
+	// Get the group from Tharsis.
+	found, err := t.client.Group.GetGroup(ctx, &ttypes.GetGroupInput{
+		ID:   byID,
+		Path: byPath,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading group",
+			err.Error(),
+		)
+		return
+	}
+
+	if found == nil {
+
+		if byPath != nil {
+			// An attempt to import by path has failed to find the specified resource.  Error out.
+			if err != nil {
+				resp.Diagnostics.AddError(
+					"Group not found: "+*byPath,
+					err.Error(),
+				)
+				return
+			}
+		}
+
+		// Handle the case that the group no longer exists if that fact is reported by returning nil.
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	// Copy the from-Tharsis struct to the state.
+	t.copyGroup(*found, &state)
+
+	// Set the refreshed state, whether or not there is an error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (t *groupResource) Update(ctx context.Context,
+	req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	// Retrieve values from plan.
+	var plan GroupModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update the group via Tharsis.
+	// The ID is used to find the record to update.
+	// The description is modified.
+	updated, err := t.client.Group.UpdateGroup(ctx,
+		&ttypes.UpdateGroupInput{
+			ID:          ptr.String(plan.ID.ValueString()),
+			Description: plan.Description.ValueString(),
+		})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating group",
+			err.Error(),
+		)
+		return
+	}
+
+	// Copy all fields returned by Tharsis back into the plan.
+	t.copyGroup(*updated, &plan)
+
+	// Set the response state to the fully-populated plan, with or without error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (t *groupResource) Delete(ctx context.Context,
+	req resource.DeleteRequest, resp *resource.DeleteResponse) {
+
+	// Get the current state.
+	var state GroupModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete the group via Tharsis.
+	err := t.client.Group.DeleteGroup(ctx,
+		&ttypes.DeleteGroupInput{
+			ID: ptr.String(state.ID.ValueString()),
+		})
+	if err != nil {
+
+		// Handle the case that the group no longer exists.
+		if t.isErrorGroupNotFound(err) {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+
+		resp.Diagnostics.AddError(
+			"Error deleting group",
+			err.Error(),
+		)
+	}
+}
+
+// ImportState helps the provider implement the ResourceWithImportState interface.
+func (t *groupResource) ImportState(ctx context.Context,
+	req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+
+	// Import by full path.
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("full_path"), req.ID)...)
```

Comment:
Based on the docs I think the conventional way to handle an import by a non-id field would be to make the SDK call to get the resource in this function and then set the state the same way the Read function does. That'll allow the read function to always assume that an ID is present in the state.